rs/ic_os/config/src/lib.rs

```diff
@@ -185,59 +154,270 @@ pub fn get_config_ini_settings(config_file_path: &Path) -> Result<(NetworkSettin
+        };
+        let setupos_settings = SetupOSSettings;
+        let hostos_settings = HostOSSettings {
+            vm_memory: 490,
+            vm_cpu: "kvm".to_string(),
+            verbose: false,
+        };
+        let guestos_settings = GuestOSSettings {
+            ic_crypto_path: None,
+            ic_state_path: None,
+            ic_registry_local_store_path: None,
+            guestos_dev: GuestosDevConfig::default(),
+        };
+
+        let setupos_config_struct = SetupOSConfig {
+            network_settings: network_settings.clone(),
+            icos_settings: icos_settings.clone(),
+            setupos_settings: setupos_settings.clone(),
+            hostos_settings: hostos_settings.clone(),
+            guestos_settings: guestos_settings.clone(),
+        };
+        let hostos_config_struct = HostOSConfig {
+            network_settings: network_settings.clone(),
+            icos_settings: icos_settings.clone(),
+            hostos_settings: hostos_settings.clone(),
+            guestos_settings: guestos_settings.clone(),
+        };
+        let guestos_config_struct = GuestOSConfig {
+            network_settings: network_settings.clone(),
+            icos_settings: icos_settings.clone(),
+            guestos_settings: guestos_settings.clone(),
+        };
+
+        let temp_dir = tempdir().expect("Failed to create temp directory");
+        let json_setupos_file = temp_dir.path().join("test_setupos_config.json");
+        let json_hostos_file = temp_dir.path().join("test_hostos_config.json");
+        let json_guestos_file = temp_dir.path().join("test_guestos_config.json");
+
+        assert!(serialize_and_write_config(&json_setupos_file, &setupos_config_struct).is_ok());
+        assert!(serialize_and_write_config(&json_hostos_file, &hostos_config_struct).is_ok());
+        assert!(serialize_and_write_config(&json_guestos_file, &guestos_config_struct).is_ok());
+
+        let deserialized_setupos_config: SetupOSConfig =
+            deserialize_config(json_setupos_file.to_str().unwrap())
+                .expect("Failed to deserialize setupos config");
+        let deserialized_hostos_config: HostOSConfig =
+            deserialize_config(json_hostos_file.to_str().unwrap())
+                .expect("Failed to deserialize hostos config");
+        let deserialized_guestos_config: GuestOSConfig =
+            deserialize_config(json_guestos_file.to_str().unwrap())
+                .expect("Failed to deserialize guestos config");
+
+        assert_eq!(setupos_config_struct, deserialized_setupos_config);
+        assert_eq!(hostos_config_struct, deserialized_hostos_config);
+        assert_eq!(guestos_config_struct, deserialized_guestos_config);
+    }
+
+    #[test]
+    fn test_is_valid_ipv6_prefix() {
+        assert!(is_valid_ipv6_prefix("2a00:1111:1111:1111"));
+        assert!(is_valid_ipv6_prefix("2a00:111:11:11"));
+        assert!(is_valid_ipv6_prefix("2602:fb2b:100:10"));
```

Comment:
Consider adding a few examples that are not valid prefixes (bonus if not trivial examples)

Reply:
Added!